### ISSUE TITLE
`Failed to create folder root directory mkdir ...: operation not permitted`

### ISSUE THREAD

**Issue by hwiorn** [MobiusSync/MobiusSync#106]
Recently, I removed every device and folder on my iPhone, iPad and my laptop. And then I reorganized my folders and started to sync again. After device was connected on iPhone and iPad, I got  `Failed to create folder root directory mkdir ...: operation not permitted` error. It seems similar to the #66 issue. 

Fortunately, the workaround was changing the folder labels to something other temporarily, such as `Books` to `Books1`.

#### iPhone log
```
2023-03-22 21:21:02: Failed to create folder root directory mkdir /private/var/mobile/Containers/Data/Application/768724E2-1290-45C5-9081-138C7B2F2596/jabref-library: operation not permitted
2023-03-22 21:21:02: Error on folder "jabref-library" (amqeh-tdnfz) at ~/jabref-library: folder path missing
```

#### iPad log
```
2023-03-22 21:23:45: Failed to create folder root directory mkdir /private/var/mobile/Containers/Data/Application/2AAC4DC3-D97B-4A8C-835A-A9095E6D09EB/jabref-library: operation not permitted

2023-03-22 21:23:45: Failed to create folder root directory mkdir /private/var/mobile/Containers/Data/Application/2AAC4DC3-D97B-4A8C-835A-A9095E6D09EB/Books: operation not permitted

2023-03-22 21:23:45: Error on folder "jabref-library" (amqeh-tdnfz) at ~/jabref-library: folder path missing

2023-03-22 21:23:45: Error on folder "Books" (gywg5-r33ma) at ~/Books: folder path missing
```

**Comment by pruszewski** [comment]
I have exacly same problem. 
I have bought the app, then tried to connect it to my existing syncthing devices setup (with only one folder named: NEPTUN as a share) and I'm getting the same error.

**Comment by mikolajblog** [comment]
I have exacly same problem.
I have bought the app, then tried to connect it to my existing syncthing devices setup (with only one folder named: NEPTUN as a share) and I'm getting the same error.

**Comment by SimonPickup** [comment]
Does this happen after a reinstall of the app, or only after creating, deleting, recreating shares? Need more details to be able to reproduce.

**Comment by mikolajblog** [comment]
It was a fresh install. I've installed the app for the first time.

**Comment by hwiorn** [comment]
> Does this happen after a reinstall of the app, or only after creating, deleting, recreating shares? Need more details to be able to reproduce.

In my case, only after deleting everything, then recreating (same labels), the issue was happened. 

I'm syncing my books and its annotations between Linux server(for hub), Linux laptop, e-book reader(android), iPhone and iPad.

I met this issue only on iOS devices.

**Comment by Jesus-M** [comment]
Same problem here.
Changes to a new phone in April, tried to install it and got the error, no possible to sync.
In the old phone it was working fine.

**Comment by DefaultGen** [comment]
I hit this same issue after attempting to rename my shares to change the capitalization. I had existing shares with Folder Labels of Sync1, Sync2, and Sync3 and Folder Paths of ./Sync1, ./Sync2, and /Sync3. I deleted these from Mobius Sync then reshared them from another computer running Syncthing with the names sync1, sync2, and sync3. Mobius Sync adds the folders but doesn't sync them and shows the " Failed to create folder root directory mkdir ...: operation not permitted" error.

The above workaround worked. I changed the Folder Labels to something temporary by adding a character: sync1x, sync2x, and sync3x and saved. After this, they synced correctly. Afterwards, I was able to change them back to the labels I wanted (sync1, sync2, sync3) and it seems to be working fine now.
